### PR TITLE
feat: pack support for vehicles and exotic beasts

### PR DESCRIPTION
**Pull request by tgvashworth**
## Summary

- Pack authors can create VEHICLE and EXOTIC_BEAST fighters in a content pack. The form auto-picks the matching `ContentStatlineType` (Vehicle / Fighter respectively).
- Each one auto-spawns a read-only companion `ContentEquipment` ("Vehicles" / "Status Items") + `ContentEquipmentFighterProfile` bridge + `CustomContentPackItem`. Cost mirrors `fighter.base_cost` and stays in sync via a post_save signal. The companion is linked to its source fighter by an explicit `auto_companion_for_fighter` `OneToOneField` on `ContentEquipment` — the canonical key for all auto-companion lookups.
- Subscribed lists can buy these via the existing flows (vehicle picker for vehicles, equipment picker for beasts) — `VehicleSelectionForm` is now pack-aware. Archive/restore of a pack vehicle/beast fighter cascades to the companion's pack item.

## Interim block on default-assignments

Pack authors **cannot** add a fighter-linked equipment (vehicle / exotic beast) as a default assignment for a pack fighter. Today such a default would only materialise on newly-hired list-fighters in subscribed gangs — existing list-fighters of that type would silently miss the child. The default-equipment picker (gear and weapon) excludes equipment with a `ContentEquipmentFighterProfile` until issue #1725 lands the retroactive-propagation work.

The underlying spawn-on-hire mechanism is intact and tested via ORM-direct creation (`test_default_vehicle_assignment_spawns_child_when_fighter_hired`) so once the propagation work lands the picker can be re-opened with no further code changes.

## Test plan

- [ ] Create a VEHICLE pack fighter (e.g. "Goliath Mauler"); verify the fighter card shows "Available as equipment 'Goliath Mauler' for 150¢" and the auto-equipment row in the gear section has no Edit/Archive links
- [ ] Edit the fighter's base_cost; verify the linked equipment cost updates
- [ ] Create an EXOTIC_BEAST pack fighter; verify the auto-equipment is in the "Status Items" category
- [ ] Subscribe a list to the pack; verify the vehicle appears in the vehicle picker
- [ ] Buy the vehicle through the 3-step flow; verify a child fighter (vehicle) is spawned
- [ ] Buy the beast through the regular equipment picker on a fighter; verify the beast child fighter is spawned
- [ ] Archive the pack vehicle fighter; verify the companion equipment row is also hidden from the pack detail page
- [ ] Restore the pack vehicle fighter; verify the companion is restored
- [ ] Verify that VEHICLE / EXOTIC_BEAST appear in the pack create-fighter category dropdown; STASH and GANG_TERRAIN do not
- [ ] Verify the pack default-equipment picker does NOT list any vehicle/beast equipment

## Related

- Issue #1725 — async retroactive-propagation work that will lift the default-assignment block

## Next steps

- Manual smoke-test the vehicle 3-step purchase flow in a browser

### CODE REVIEW

File: gyrinx/core/views/pack.py [hunk 1]
```diff
@@ -1110,6 +1145,82 @@ def _create_fighter_statline(
...
+        name=fighter.type,
+        category=category,
+        cost=str(fighter.base_cost),
+    )
+    equipment._history_user = user
+    equipment.save()
+
+    ContentEquipmentFighterProfile.objects.create(
+        equipment=equipment, content_fighter=fighter
+    )
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply gyrinx/core/views/pack.py]: Fixed in 6467d83b — added an explicit `auto_companion_for_fighter` OneToOneField on `ContentEquipment` (migration 0163). The OneToOne enforces 1:1 at the DB level, and `_ensure_auto_equipment_for_fighter` now looks up by FK so the lookup is deterministic across all scenarios.

File: gyrinx/content/models/signal_handlers.py [hunk 2]
```diff
@@ -513,6 +517,38 @@ def create_fighter_cost_action(sender, instance, created, **kwargs):
...
+    equipment = bridge.equipment
+    changed = False
+    if equipment.name != instance.type:
+        equipment.name = instance.type
+        changed = True
+    if equipment.cost != str(instance.base_cost):
+        equipment.cost = str(instance.base_cost)
+        changed = True
+    if changed:
+        equipment.save()
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply gyrinx/content/models/signal_handlers.py]: Fixed in 6467d83b — `sync_auto_equipment_cost` now (a) short-circuits when the fighter category isn't in {VEHICLE, EXOTIC_BEAST}, (b) looks up via the new `auto_companion_for_fighter` FK so the target is unambiguous, and (c) syncs equipment category as well as name+cost (covers VEHICLE↔EXOTIC_BEAST conversion).

File: gyrinx/core/views/pack.py [hunk 3]
```diff
@@ -1110,6 +1145,82 @@ def _create_fighter_statline(
...
+        if equipment.cost != str(fighter.base_cost):
+            equipment.cost = str(fighter.base_cost)
+            changed = True
+        if equipment.category_id != category.pk:
+            equipment.category = category
+            changed = True
+        if changed:
+            equipment._history_user = user
+            equipment.save()
+        return equipment
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply gyrinx/core/views/pack.py]: Fixed in 6467d83b — `_ensure_auto_equipment_for_fighter` now looks up the companion via the new `auto_companion_for_fighter` FK (deterministic, OneToOne-enforced). The existing-bridge path also re-registers the pack item if it has gone missing, addressing the second half of your point.

File: gyrinx/core/views/pack.py [hunk 4]
```diff
@@ -683,6 +687,33 @@ def get_context_data(self, **kwargs):
...
+        auto_equipment_by_fighter_id = {}
+        auto_equipment_ids = set()
+        if pack_equipment_ids:
+            for bridge in ContentEquipmentFighterProfile.objects.filter(
+                equipment_id__in=pack_equipment_ids
+            ).select_related("equipment", "content_fighter"):
+                auto_equipment_by_fighter_id[bridge.content_fighter_id] = (
+                    bridge.equipment
+                )
+                auto_equipment_ids.add(bridge.equipment_id)
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply gyrinx/core/views/pack.py]: Fixed in 6467d83b — the pack-detail `is_auto_equipment` flag is now keyed off `auto_companion_for_fighter__isnull=False` rather than "any ContentEquipmentFighterProfile bridge". Manually-authored equipment with a bridge is no longer mis-classified as read-only.

File: gyrinx/core/forms/pack.py [hunk 5]
```diff
@@ -20,10 +20,17 @@
...
-    FighterCategoryChoices.VEHICLE,
     FighterCategoryChoices.GANG_TERRAIN,
+}
+
+# Categories whose fighters are spawned via a ContentEquipmentFighterProfile
+# bridge — the pack flow auto-creates the equipment + bridge + pack item.
+_AUTO_EQUIPMENT_FIGHTER_CATEGORIES = {
+    FighterCategoryChoices.VEHICLE,
     FighterCategoryChoices.EXOTIC_BEAST,
 }
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply gyrinx/core/forms/pack.py]: Fixed in 6467d83b — constant deleted; the auto-equipment behaviour is keyed off `AUTO_EQUIPMENT_CATEGORY_BY_FIGHTER_CATEGORY` in `content/models/equipment.py`.

File: gyrinx/core/tests/test_pack_vehicles_beasts.py [hunk 6]
```diff
@@ -0,0 +1,659 @@
...
+        base_cost="100",
+    )
+    # Either the form rejects (status 200 with error) OR it redirects but
+    # leaves no statline → in either case the fighter MUST NOT be created
+    # with a default fighter statline.
+    assert (
+        not ContentFighter.objects.all_content()
+        .filter(type="Bare Vehicle", custom_statline__statline_type__name="Fighter")
+        .exists()
+    )
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply gyrinx/core/tests/test_pack_vehicles_beasts.py]: Fixed in 6467d83b — test now drives Step 2 too. While verifying, also caught a real correctness bug: `_get_statline_type_for_category` was silently falling back to the Fighter statline for VEHICLE when no Vehicle type existed (producing wrong stats). Now raises ContentStatlineType.DoesNotExist for auto-equipment categories. Test asserts no fighter is created.

File: .claude/notes/pack-vehicles-beasts-plan.md [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+  cost.
+- The signal at `list.py:4604` raises `ValueError` if multiple bridges
+  exist for one equipment. Our auto-create flow creates one bridge per
+  equipment — keep it that way.
+- If a pack fighter is archived/deleted, the equipment + bridge should
+  follow. CASCADE on `ContentEquipmentFighterProfile.equipment` is set,
+  so deleting the equipment cascades the bridge. Need to delete the
+  equipment when the fighter is deleted (or rely on `Content` archiving).
+- Migration risk: zero — we're not changing any schema. All net-new
+  records follow existing schemas.
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply .claude/notes/pack-vehicles-beasts-plan.md]: Fixed in fee4c47d — note updated. The plan was originally schema-clean but PR feedback round 1 surfaced the need for an explicit FK link, so migration 0163 was added.

File: gyrinx/core/views/pack.py [hunk 8]
```diff
@@ -683,6 +688,39 @@ def get_context_data(self, **kwargs):
             else:
                 active_by_slug[slug].append(entry_data)
 
+        # Build a map of equipment_id → (auto_fighter, profile_id) for any
+        # ContentEquipmentFighterProfile bridges that touch equipment in this
+        # pack. Used below to:
+        #   - mark gear/weapon entries that are auto-generated (so the
+        #     template can suppress edit/archive links)
+        #   - annotate fighter entries with their linked auto-equipment so
+        #     the fighter card can render "Available as equipment X for Y¢".
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply gyrinx/core/views/pack.py]: Fixed in fee4c47d — comment was indeed stale from the bridge-based version. Updated to describe `auto_equipment_by_fighter_id` and `auto_equipment_ids` (and the FK keying).

File: gyrinx/core/templates/core/pack/includes/fighter_preview_card.html [hunk 9]
```diff
@@ -29,6 +29,7 @@ <h3 class="h5 mb-0">{{ content_fighter.type }}</h3>
             <span class="text-secondary fs-7">
                 {{ content_fighter.get_category_display }}
                 {% if content_fighter.house %}· {{ content_fighter.house.name }}{% endif %}
+                {% if auto_equipment %}· Available as equipment "{{ auto_equipment.name }}" for {{ auto_equipment.cost }}¢{% endif %}
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply gyrinx/core/templates/core/pack/includes/fighter_preview_card.html]: Fixed in fee4c47d — switched to `{{ auto_equipment.cost_display }}` from CostMixin so formatting is consistent with the rest of the UI.

File: gyrinx/core/views/pack.py [hunk 10]
```diff
@@ -2807,19 +2979,31 @@ def _get_pack_fighter(pack, item_id):
...
     Returns a queryset of ContentEquipment filtered to the same category
     groups that pack creators can use when adding gear/weapons, with
     weapon profiles correctly loaded (including pack profiles).
+
+    ``exclude_fighter_linked`` filters out equipment with a
+    ``ContentEquipmentFighterProfile`` (i.e. vehicle / exotic-beast
+    spawners). The pack-fighter default-assignment picker passes True so
+    pack authors can't silently configure a default that would only
+    apply to gangs hired AFTER the change. See issue #1725 for the
+    follow-up retroactive-propagation work that will lift this block.
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply gyrinx/core/views/pack.py:2997]: Acknowledged — PR description updated. The default-assignment block is intentional pending retroactive-propagation work in #1725. Without retroactive propagation, adding a default would only materialise for newly-hired list-fighters and silently miss existing ones, so we block at the picker until #1725 lands. The underlying spawn-on-hire mechanism is intact and tested via ORM-direct creation.

File: gyrinx/content/models/equipment.py [hunk 11]
```diff
@@ -433,6 +444,20 @@ class UpgradeMode(models.TextChoices):
...
+        null=True,
+        blank=True,
+        related_name="auto_companion_equipment",
+        help_text=(
+            "Set only by the pack vehicle/exotic-beast auto-create flow. "
+            "This equipment is the read-only companion that lets list-buyers "
+            "purchase the linked fighter. Manually-authored equipment must "
+            "leave this NULL."
+        ),
+    )
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply gyrinx/content/models/equipment.py:459]: Acknowledged but not implementing in this PR. The trait/accessory `save() → full_clean()` pattern fits those models because they have `validate_unique` logic that needs enforcement on direct ORM writes. ContentEquipment's `clean()` only strips whitespace — adding the override breaks 144 existing tests that create ContentEquipment without a category (DB allows NULL via `null=True` but `blank=False` makes full_clean reject). The cost-to-benefit is unfavourable here. Will revisit if a real data-integrity issue surfaces; happy to take it as a separate refactor that updates the test fixtures.
